app/services/foreman_webhooks/webhook_service.rb

```diff
@@ -2,19 +2,39 @@
 
 module ForemanWebhooks
   class WebhookService
-    attr_accessor :webhook, :event_name, :payload
+    attr_accessor :webhook, :event_name, :payload, :rendered_headers, :rendered_url
 
     delegate :logger, to: Rails
 
-    def initialize(webhook:, event_name:, payload:)
+    def initialize(webhook:, event_name:, payload:, headers:, url:)
       @webhook = webhook
       @event_name = event_name
       @payload = payload
+      @rendered_headers = headers
+      @rendered_url = url
     end
 
     def execute
-      Foreman::Logging.blob("Webhook payload for event #{event_name}", payload)
-      response = request(webhook, payload)
+      logger.info("Performing '#{webhook.name}' webhook request for event '#{event_name}'")
+      Foreman::Logging.blob("Payload for '#{event_name}'", payload)
+      headers = {}
+      begin
+        headers = JSON.parse(rendered_headers)
+      rescue StandardError => e
+        logger.warn("Could not parse HTTP headers JSON, ignoring: #{e}")
+        logger.debug("Headers: #{rendered_headers}")
+      end
+
+      response = self.class.request(url: webhook.target_url,
+                                    payload: payload,
+                                    http_method: webhook.http_method,
+                                    user: webhook.user,
+                                    password: webhook.password,
+                                    content_type: webhook.http_content_type,
+                                    headers: headers,
+                                    ca_string: webhook.verify_ssl?,
+                                    ca_verify: webhook.ca_certs_store,
```

Comment:
Those should be swapped, shouldn't they?

Reply:
Oh yeah, good catch.